Deadlock.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
```

Comment:
Consider marking the `myLock1` and `myLock2` objects as `readonly` to ensure their values do not change after initialization. This will enhance the thread safety of your code.